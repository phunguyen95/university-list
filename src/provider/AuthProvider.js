import React, { useState, useEffect } from "react";
import { signUp, logout, signIn } from "..//firestore";
import useInputState from "../hooks/useInputState";
import useDebounce from "../hooks/useDebounce";
import _isEmpty from "lodash/isEmpty";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
export const firebaseAuth = React.createContext();

const AuthProvider = (props) => {
  const initState = { email: "", password: "" };
  const [errors, setErrors] = useState([]);
  const [token, setToken] = useState(null);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    if (errors.length > 0) {
      toast.warn("Sorry,there's unexpected error, please try again later", {});
      setIsLoading(false);
      setErrors([]);
    }
  }, [errors]);

  const handleSignup = (email, password) => {
    // calling signup from firebase server
    signUp(email, password, setErrors, setToken);
  };
  const handleSignIn = (email, password) => {
    signIn(email, password, setErrors, setToken);
  };
  const handleSignout = () => {
    logout(setErrors, setToken);
    toast.info("You have been log out successfully", {});
  };

  return (
    <firebaseAuth.Provider
      value={{
        handleSignup,
        errors,
        token,
        handleSignout,
        handleSignIn,
        isLoading,
        setIsLoading,
        setErrors,
      }}
    >
      <ToastContainer
        position="top-right"
        autoClose={3000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />

      {props.children}
    </firebaseAuth.Provider>
  );
};
export default AuthProvider;
