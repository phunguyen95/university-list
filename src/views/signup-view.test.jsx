import React from 'react';
import { render, screen } from "@testing-library/react";
import AuthProvider from "../provider/AuthProvider";
import SignupView from "./signup-view";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
const useStateStub = jest.fn()
jest.spyOn(React,'useEffect').mockImplementationOnce(f=>f());
const mockUseState = () => {
    const setEmail = jest.fn();
    const setPassword = jest.fn();
    const setIsEmailInvalid = jest.fn();
    const setIsPasswordInvalid = jest.fn();
    const setIsChecked = jest.fn();

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce((_) => [_, setEmail])
      .mockImplementationOnce((_) => [_, setPassword])
      .mockImplementationOnce((_) => [_, setIsEmailInvalid])
      .mockImplementationOnce((_) => [_, setIsPasswordInvalid])
      .mockImplementationOnce((_) => [_, setIsChecked]);
}
test("renders signup view without crashing", () => {
  const history = createMemoryHistory();
  mockUseState();
  const handleSignup = jest.fn();
  const isLoading = false;
  const setIsLoading = jest.fn();
  const setErrors = jest.fn();
  const token = "";
  const { asFragment } = render(
    <Router history={history}>
      <AuthProvider value={{ handleSignup, isLoading,setIsLoading,setErrors,token }}>
        <SignupView />
      </AuthProvider>
    </Router>
  );

  expect(asFragment()).toMatchSnapshot();
});
