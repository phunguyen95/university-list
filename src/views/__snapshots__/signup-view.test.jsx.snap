// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signup view without crashing 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="signup-wrapper"
  >
    <div
      class="content"
    >
      <h4>
        REGISTER FORM
      </h4>
    </div>
    <div
      class="field"
    >
      <label
        class="label"
      >
        Email
      </label>
      <div
        class="control has-icons-left has-icons-right"
      >
        <input
          class="input"
          placeholder="Please input your email"
          type="email"
          value=""
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-envelope"
          />
        </span>
      </div>
    </div>
    <div
      class="field"
    >
      <p
        class="control has-icons-left"
      >
        <input
          class="input"
          placeholder="Minimum eight characters, at least one letter, one number and one special character"
          type="password"
          value=""
        />
        <span
          class="icon is-small is-left"
        >
          <i
            class="fas fa-lock"
          />
        </span>
      </p>
    </div>
    <div
      class="field"
    >
      <div
        class="control"
      >
        <label
          class="checkbox"
        >
          <input
            type="checkbox"
            value="false"
          />
           I agree to the
          <a
            href="#"
          >
            terms and conditions
          </a>
        </label>
      </div>
    </div>
    <div
      class="field is-grouped"
    >
      <div
        class="control"
      >
        <button
          class="button is-link"
          disabled=""
        >
          Register
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
